apps/src/lib/util/javascriptMode.js

```diff
@@ -64,25 +65,32 @@ export function apiValidateType(
           typeof varValue === 'boolean';
         break;
       case 'pinid':
-        var validPins = [
-          'A0',
-          'A1',
-          'A2',
-          'A3',
-          'A4',
-          'A5',
-          'A6',
-          'A7',
-          0,
-          1,
-          2,
-          3,
-          6,
-          9,
-          10,
-          12
-        ];
-        var reservedPins = ['A2', 'A3', 'A7', 1, 9, 10];
+        var validPins, reservedPins;
```

Comment:
nit: you could default assign these to either the CircuitPlayground/Microbit values and then override them if needed, i.e.
```
var validPins = [A0, A1, ...]
var reservedPins = [A2, A3 ...]
if (isMicroBitBoard) {
   validPins = [0, 1, 2] ...
}
```

Reply:
Sure - will do!